freqtrade/constants.py

```diff
@@ -99,380 +145,414 @@
                     },
-                    'uniqueItems': True
+                    "uniqueItems": True,
                 },
-                'pair_blacklist': {
-                    'type': 'array',
-                    'items': {
-                        'type': 'string',
+                "pair_blacklist": {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
                     },
-                    'uniqueItems': True
+                    "uniqueItems": True,
                 },
-                'unknown_fee_rate': {'type': 'number'},
-                'outdated_offset': {'type': 'integer', 'minimum': 1},
-                'markets_refresh_interval': {'type': 'integer'},
-                'ccxt_config': {'type': 'object'},
-                'ccxt_async_config': {'type': 'object'}
+                "unknown_fee_rate": {"type": "number"},
+                "outdated_offset": {"type": "integer", "minimum": 1},
+                "markets_refresh_interval": {"type": "integer"},
+                "ccxt_config": {"type": "object"},
+                "ccxt_async_config": {"type": "object"},
+            },
+            "required": ["name"],
+        },
+        "edge": {
+            "type": "object",
+            "properties": {
+                "enabled": {"type": "boolean"},
+                "process_throttle_secs": {"type": "integer", "minimum": 600},
+                "calculate_since_number_of_days": {"type": "integer"},
+                "allowed_risk": {"type": "number"},
+                "stoploss_range_min": {"type": "number"},
+                "stoploss_range_max": {"type": "number"},
+                "stoploss_range_step": {"type": "number"},
+                "minimum_winrate": {"type": "number"},
+                "minimum_expectancy": {"type": "number"},
+                "min_trade_number": {"type": "number"},
+                "max_trade_duration_minute": {"type": "integer"},
+                "remove_pumps": {"type": "boolean"},
             },
-            'required': ['name']
+            "required": ["process_throttle_secs", "allowed_risk"],
         },
-        'edge': {
-            'type': 'object',
-            'properties': {
-                'enabled': {'type': 'boolean'},
-                'process_throttle_secs': {'type': 'integer', 'minimum': 600},
-                'calculate_since_number_of_days': {'type': 'integer'},
-                'allowed_risk': {'type': 'number'},
-                'stoploss_range_min': {'type': 'number'},
-                'stoploss_range_max': {'type': 'number'},
-                'stoploss_range_step': {'type': 'number'},
-                'minimum_winrate': {'type': 'number'},
-                'minimum_expectancy': {'type': 'number'},
-                'min_trade_number': {'type': 'number'},
-                'max_trade_duration_minute': {'type': 'integer'},
-                'remove_pumps': {'type': 'boolean'}
+        "freqai": {
```

Comment:
@xmatthias can you check this? I added `config['freqai']` to constants in an effort to pass the unittests, but still doesnt seem to pass:

https://github.com/freqtrade/freqtrade/runs/6307348305?check_suite_focus=true#step:8:7483

Reply:
(also accidentally black formatted the file, but it doesnt appear to cause any bugs here, we can revert it if you prefer)

Reply:
yeah - for now i'd ask you to revert it. the file should format fine with autopep8 - black changes too much for a PR like this - making Reviews near impossible (almost every line in this file is changed (mostly due to quotes) - which is an irrelevant change for this PR - but it took me near 2 minutes to reach the things you actually changed (the things i SHOULD be looking at).

Reply:
Reverted, and added changes on top.